### PR TITLE
[OT171-40]  Testimonials integration

### PR DESCRIPTION
**Pull request by frannasich**
- TestimonialsList, TestimonialsResponse and TestimonialsService added
- override deleted
- View model updated with testimonials logic
- cellForItemAt and didSelected Testimonials updated
- Fixed some errors of navigation in Splash
- Add condition to Hide Testimonials section

**comments:** Splash contains errors... LogInViewController is not show anymore

https://user-images.githubusercontent.com/80861248/162098257-716643b2-f4d7-489c-b3be-3d2158e7c36d.mov

### Jira Ticket
[OT171-40](https://alkemy-labs.atlassian.net/jira/software/c/projects/OT171/boards/247?modal=detail&selectedIssue=OT171-40)

### CODE REVIEW

**Review by gapoj — CHANGES_REQUESTED** [review]
Simulator Screen Shot - iPad Pro (9 7-inch) - 2022-04-08 at 00 56 38
the story saids "En caso que el listado sea vacío, se deberá ocultar dicha sección" but when you don't have data it looks that the screenshot above

**Comment by gapoj** [comment]
another thing you are not actually showing the testimony only the picture and the name of the person giving the testimony, this is a detail but since you are already will have to fix things

**Comment by frannasich** [comment]
> another thing you are not actually showing the testimony only the picture and the name of the person giving the testimony, this is a detail but since you are already will have to fix things

image

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 1]
```diff
@@ -81,30 +74,43 @@ extension HomeViewController: UICollectionViewDataSource, UICollectionViewDelega
...
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         switch collectionView{
         case testimonialsCollectionView:
             // Add seeMore page
-            if indexPath.row == min(testimonialsData.count, 4) {
+            if indexPath.row == min(sliderViewModel!.getTestimonialsCount(), 4) {
                 let cell = testimonialsCollectionView.dequeueReusableCell(withReuseIdentifier: "seeMoreCell", for: indexPath) as? SeeMoreCollectionViewCell
                 
                 return cell ?? SeeMoreCollectionViewCell()
+            } else if indexPath.row == sliderViewModel?.getTestimonialsCount() && sliderViewModel?.getTestimonialsCount() == 0 {
```
[the remark this answers is not in the document]
Comment (gapoj) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:99]: - please take out this 
-  read the documentation so you understand when func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell  is called
- you should hide the entire section
- if you already are hiding the section be careful of the empty space, should be reduced 
- if yo have so many lines of code in a case you should separate this in a function

**Review by gapoj — APPROVED** [review]

**Pull request merged**